Description: Fixing https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=846749
 After upgrading to SqlAlchemy >= 1.1.0, new instances of worker model get 
 id=None after save method invocation. However id is primary_key.
Forwarded: https://bugs.launchpad.net/cinder/+bug/1651448
Author: Ondřej Kobližek <koblizeko@gmail.com>

--- a/cinder/db/sqlalchemy/models.py
+++ b/cinder/db/sqlalchemy/models.py
@@ -847,9 +847,9 @@
 
     # Type of the resource we are working on (Volume, Snapshot, Backup) it must
     # match the Versioned Object class name.
-    resource_type = Column(String(40), primary_key=True, nullable=False)
+    resource_type = Column(String(40), index=True, nullable=False)
     # UUID of the resource we are working on
-    resource_id = Column(String(36), primary_key=True, nullable=False)
+    resource_id = Column(String(36), index=True, nullable=False)
 
     # Status that should be cleaned on service failure
     status = Column(String(255), nullable=False)
